docker_mcp/services/cleanup.py

```
@@ -732,98 +722,115 @@ def _calculate_cleanup_levels(self, summary: dict) -> dict[str, Any]:
+                "docker",
+                "ps",
+                "-a",
+                "--filter",
+                "status=exited",
+                "--format",
+                "{{.Names}}",
             ]
             containers_result = await asyncio.get_event_loop().run_in_executor(
                 None,
-                lambda: subprocess.run(containers_cmd, check=False, capture_output=True, text=True)  # nosec B603
+                lambda: subprocess.run(containers_cmd, check=False, capture_output=True, text=True),  # nosec B603
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)